invokeai/backend/model_manager/load/model_cache/model_cache_default.py

```diff
@@ -267,6 +268,13 @@ def move_model_to_device(self, cache_entry: CacheRecord[AnyModel], target_device
         start_model_to_time = time.time()
         snapshot_before = self._capture_memory_snapshot()
         try:
+            if target_device == self.storage_device:
+                cache_entry.model.load_state_dict(cache_entry.state_dict, assign=True)
+            else:
+                new_dict: Dict[str, torch.Tensor] = {}
+                for k, v in cache_entry.state_dict.items():
+                    new_dict[k] = v.to(torch.device(target_device), copy=True)
+                cache_entry.model.load_state_dict(new_dict, assign=True)
```

Comment:
Currently, the `cache_entry.state_dict` could be `None`, because it is initialized with `state_dict = model.state_dict() if hasattr(model, "state_dict") else None`. We need to handle that case here.

It might be better to check `isinstance(model, torch.nn.Module)` instead of `hasattr(model, "state_dict")`, since we also need it to have the `.load_state_dict(...)` method.

Reply:
Fixed.